// @sap/cloud-sdk-core/dist/scp-cf/destination-cache.js
"use strict";
/*!
 * Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved.
 */
Object.defineProperty(exports, "__esModule", { value: true });
var util_1 = require("../util");
var cache_1 = require("./cache");
// TODO: make class, deprecate function, check if I correctly see this as user
var DestinationCache = function (cache) { return ({
    retrieveDestinationFromCache: function (decodedJwt, name, isolation) {
        return cache.get(getDestinationCacheKey(decodedJwt, name, isolation));
    },
    // TODO: the type of the second parameter does not make any sense, since it's flattened anyway...
    // TODO: isolation should be optional, since the switch statement has a default case which would handle undefined
    cacheRetrievedDestinations: function (decodedJwt, retrievedDestinations, isolation) {
        retrievedDestinations.instance.concat(retrievedDestinations.subaccount).forEach(function (destination) {
            if (!destination.name) {
                throw new Error('The destination name is undefined.');
            }
            var key = getDestinationCacheKey(decodedJwt, destination.name, isolation);
            cache.set(key, destination);
        });
    },
    clear: function () {
        cache.clear();
    },
    getCacheInstance: function () {
        return cache;
    }
}); };
/**
 * Calculates a cache key based on the jwt and destination name for the given isolation strategy.
 * Cache keys for strategies are non-overlapping, i.e. using a cache key for strategy [[IsolationStrategy.Tenant]]
 * will not result in a cache hit for a destination that has been cached with strategy [[IsolationStrategy.Tenant_User]].
 *
 * @param decodedJwt The decoded JWT of the current request.
 * @param destinationName The name of the destination.
 * @param isolationStrategy The strategy used to isolate cache entries.
 * @returns The cache key.
 * @hidden
 */
function getDestinationCacheKey(decodedJwt, destinationName, isolationStrategy) {
    switch (isolationStrategy) {
        case cache_1.IsolationStrategy.No_Isolation:
            return "::" + destinationName;
        case cache_1.IsolationStrategy.Tenant_User:
            return util_1.tenantId(decodedJwt) + ":" + util_1.userId(decodedJwt) + ":" + destinationName;
        case cache_1.IsolationStrategy.User:
            return ":" + util_1.userId(decodedJwt) + ":" + destinationName;
        default:
            return util_1.tenantId(decodedJwt) + "::" + destinationName;
    }
}
exports.getDestinationCacheKey = getDestinationCacheKey;
exports.destinationCache = DestinationCache(new cache_1.Cache({ hours: 0, minutes: 5, seconds: 0 }));
