// @sap/cds/lib/build/buildTaskHandlerOData.js
const path = require('path')
const BuildTaskHandler = require('./buildTaskHandler')
const { BUILD_OPTION_OUTPUT_MODE, OUTPUT_MODE_RESULT_ONLY } = require('./constants')

class BuildTaskHandlerOData extends BuildTaskHandler {
    constructor(name, task, buildOptions) {
        super(name, task, buildOptions)
        this._result = {
            dest: task.dest,
            csn: {},
            edmx: new Map(),
            languages: new Set(),
            services: new Set()
        }
    }

    async _compileCsn(model, csnDest) {
        // csn for service providers
        const csn = this.cds.compile.for.odata(model)
        this._result.csn = csn

        if (!this.hasBuildOption(BUILD_OPTION_OUTPUT_MODE, OUTPUT_MODE_RESULT_ONLY)) {
            await this.write(this.cds.compile.to.json(csn)).to(path.join(csnDest, 'csn.json'))
        }
    }

    async _compileEdmx(model, edmxDest) {
        const promises = []
        const services = this.cds.reflect(model).all(this.cds.service)
        const locales = this.cds.localize(model, this.task.options.lang || ['all'])

        if (locales) {
            for (let [localizedModel, {
                lang
            }] of locales) {
                this._result.languages.add(lang)

                if (services.length > 0) {
                    // new compile impl is throwing error in case no services exist!
                    const result = this.cds.compile.to.edmx(localizedModel, {
                        service: 'all'
                    })
                    if (result) {
                        for (let [content, key] of result) {
                            const serviceName = key.file ? key.file : key.name
                            const fileName = serviceName + (lang ? '_' + lang + '.xml' : '.xml')
                            this._result.edmx.set(fileName, content)
                            this._result.services.add(serviceName)

                            if (!this.hasBuildOption(BUILD_OPTION_OUTPUT_MODE, OUTPUT_MODE_RESULT_ONLY)) {
                                promises.push(this.write(content).to(path.join(edmxDest, fileName)))
                            }
                        }
                    }
                }
            }
        }
        return Promise.all(promises)
    }
}
module.exports = BuildTaskHandlerOData
